portal/index.php

```diff
@@ -35,7 +38,36 @@
     echo xlt('Patient Portal is turned off');
     exit;
 }
-
+$auth['portal_pwd'] = '';
+if (isset($_GET['woops'])) {
+    unset($_GET['woops']);
+    unset($_SESSION['password_update']);
+}
+if (isset($_GET['forward']) && isset($_GET['validate'])) {
+    $validate = hex2bin($_GET['validate']);
+    if ($validate <= time()) {
```

Comment:
Agree with a time contraint on a login token, but seems odd that this is not stored on server side (would be straighforward to change the validate parameter on the link and remove the time constraint).

Reply:
Another really cool way to do this (and not need to store this info on server), is to encrypt it before doing the binhex call on the validate parameter data. Then when you decrypt it here something that happens automatically in decryption is that it checks the signature of the encrypted data with the hidden key (if the signature doesn't match, then it knows that somebody tried to mess with it, and it will just return an empty field).

Reply:
I was going to do it this way but decided against. The token does not include the encoded password but is unique to a one time. I want to have a field populated to indicate a one time is out for the patient. The validate is only to check to see if the token has expired.